src/err/mod.rs

```diff
@@ -327,15 +333,70 @@ impl PyErr {
             Some(obj) => obj.as_ptr(),
         };
 
-        unsafe {
-            let null_terminated_name =
-                CString::new(name).expect("Failed to initialize nul terminated exception name");
+        let null_terminated_name =
+            CString::new(name).expect("Failed to initialize nul terminated exception name");
+
+        let ptr = unsafe {
+            ffi::PyErr_NewException(null_terminated_name.as_ptr(), base, dict)
+                as *mut ffi::PyTypeObject
+        };
+        match NonNull::new(ptr) {
+            Some(not_null) => not_null,
+            None => panic!("{}", PyErr::fetch(py)),
```

Comment:
Mmm good idea checking for errors. If we're breaking `new_type` anyway, maybe we should change it to return `PyResult<Py<PyType>>` - this pushes the error handling downstream and also makes the pointer ownership clearer?